x/poolmanager/types/params.go

```diff
@@ -11,24 +11,61 @@ import (
 
 // Parameter store keys.
 var (
-	KeyPoolCreationFee = []byte("PoolCreationFee")
+	KeyPoolCreationFee                                = []byte("PoolCreationFee")
+	KeyDefaultTakerFee                                = []byte("DefaultTakerFee")
+	KeyOsmoTakerFeeDistribution                       = []byte("OsmoTakerFeeDistribution")
+	KeyNonOsmoTakerFeeDistribution                    = []byte("NonOsmoTakerFeeDistribution")
+	KeyAdminAddresses                                 = []byte("AdminAddresses")
+	KeyCommunityPoolDenomToSwapNonWhitelistedAssetsTo = []byte("CommunityPoolDenomToSwapNonWhitelistedAssetsTo")
+	KeyAuthorizedQuoteDenoms                          = []byte("AuthorizedQuoteDenoms")
 )
 
 // ParamTable for gamm module.
 func ParamKeyTable() paramtypes.KeyTable {
 	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
 }
 
-func NewParams(poolCreationFee sdk.Coins) Params {
+func NewParams(poolCreationFee sdk.Coins,
+	defaultTakerFee sdk.Dec,
+	osmoTakerFeeDistribution, nonOsmoTakerFeeDistribution TakerFeeDistributionPercentage,
+	adminAddresses, authorizedQuoteDenoms []string,
+	communityPoolDenomToSwapNonWhitelistedAssetsTo string) Params {
 	return Params{
 		PoolCreationFee: poolCreationFee,
+		TakerFeeParams: TakerFeeParams{
+			DefaultTakerFee:                                defaultTakerFee,
+			OsmoTakerFeeDistribution:                       osmoTakerFeeDistribution,
+			NonOsmoTakerFeeDistribution:                    nonOsmoTakerFeeDistribution,
+			AdminAddresses:                                 adminAddresses,
+			CommunityPoolDenomToSwapNonWhitelistedAssetsTo: communityPoolDenomToSwapNonWhitelistedAssetsTo,
+		},
+		AuthorizedQuoteDenoms: authorizedQuoteDenoms,
 	}
 }
 
 // DefaultParams are the default poolmanager module parameters.
 func DefaultParams() Params {
 	return Params{
 		PoolCreationFee: sdk.Coins{sdk.NewInt64Coin(appparams.BaseCoinUnit, 1000_000_000)}, // 1000 OSMO
+		TakerFeeParams: TakerFeeParams{
+			DefaultTakerFee: sdk.MustNewDecFromStr("0.0015"), // 0.15%
+			OsmoTakerFeeDistribution: TakerFeeDistributionPercentage{
+				StakingRewards: sdk.MustNewDecFromStr("1"), // 100%
+				CommunityPool:  sdk.MustNewDecFromStr("0"), // 0%
+			},
+			NonOsmoTakerFeeDistribution: TakerFeeDistributionPercentage{
+				StakingRewards: sdk.MustNewDecFromStr("0.67"), // 67%
+				CommunityPool:  sdk.MustNewDecFromStr("0.33"), // 33%
+			},
+			AdminAddresses: []string{},
+			CommunityPoolDenomToSwapNonWhitelistedAssetsTo: "ibc/D189335C6E4A68B513C10AB227BF1C1D38C746766278BA3EEB4FB14124F1D858", // USDC
```

Comment:
Governance set this?

Lets in-line link to proposal if so.

(if governance didn't set this, we would do uosmo or a native denom for better compatability)

Reply:
Gov did not set this but was a discussion with Johnny as there was a desire to not have over exposure to OSMO. Can discuss on call what we should set this too

Reply:
This makes sense, Gov can set this explicitly when taker fee is turned on